### PR TITLE
samples: kscan_touch: drop the kscan_touch sample

### PR DESCRIPTION
**Pull request by fabiobaltieri**
All in-tree touchscreen drivers have been migrated over to input, the compatibility node has been removed from those boards as well, this sample can be dropped now.

### CODE REVIEW

File: samples/drivers/kscan_touch/README.rst
Comment (kartben) [inline samples/drivers/kscan_touch/README.rst:1]: can you add an entry to `doc/_scripts/redirects.py` please, pointing to whichever new location you think is most appropriate.

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/drivers/kscan_touch/README.rst
Comment (fabiobaltieri) [reply samples/drivers/kscan_touch/README.rst:1]: good idea, done, pointing to the input one

File: doc/_scripts/redirects.py [hunk 1]
```diff
@@ -163,5 +163,6 @@
     ('reference/usermode/overview', 'kernel/usermode/overview'),
     ('reference/usermode/syscalls', 'kernel/usermode/syscalls'),
     ('reference/util/index', 'kernel/util/index'),
+    ('samples/drivers/kscan_touch', 'samples/input/input'),
```
Comment (kartben) [inline doc/_scripts/redirects.py:166]:
```suggestion
    ('samples/drivers/kscan_touch', 'samples/subsys/input/input'),
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/_scripts/redirects.py (hunk 1 above)
Comment (fabiobaltieri) [reply doc/_scripts/redirects.py]: woops... fixed

**Review by faxe1008 — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Thanks! (and redirect confirmed as working now :-) )

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**